container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/ContainerTests.java

```diff
@@ -184,12 +234,36 @@ public void contractShouldBeDeployedOnlyToBesuNode() throws IOException, Transac
             goQuorumPollingTransactionReceiptProcessor,
             logValue);
 
+    int secondsWaited = 0;
+    while (!checked.get()) {
+      Thread.sleep(1000);
+      secondsWaited++;
+      if (secondsWaited > 30) {
+        fail("Waited more than 30 seconds for log.");
```

Comment:
if we fail here, does the dispose get skipped - should it be in a finally{}

Reply:
Does not really matter. The nodes are going to be stopped!